backend/lib/fetch/providers.py

```diff
@@ -84,7 +84,7 @@ def geocode_providers(providers, geocoder_name=GEOCODER, engine=connect.create_d
 
     provider_responses = []
 
-    provider_addresses = {provider['address'] for provider in providers}
+    provider_addresses = {address for address in providers}
```

Comment:
Do you actually want this to be a set in the end? I've found that often `list(set(providers))` works well -- that way indexing is still possible (note that order is lost in either case).

Reply:
I am using .difference and such so it is better to have it as a set I think